### DISCUSSION TITLE
Modifying spans annotation before training

### DISCUSSION THREAD

**Discussion by yw2903** [explosion/spaCy#6997] [Help: Coding & Implementations] [answered]
By using prodigy I am able to create a training set, however, I would like to clean the annotated entities before training with spacy (for example, removing special character at the end). However, I get error that says "span object is not modifiable
". How can I go around with that?

**Comment by svlandeg** [comment, marked as answer]
Hi!

This type of usage question is probably better suited for the new discussion forum, so I'll move it there. Don't worry if you get an email saying that this issue is closed/locked - it really is just transferred to the Discussions and will remain an open thread there.

To answer your question, I think the easiest way for you to get around this, is to query the original `doc.ents`, loop through them to clean them, and create a new (similar) set of entities that you can set afterwards with `doc.set_ents`.

**Reply by yw2903** [reply]
Thanks. However, I'm getting the following error by trying to create a span obeject 
```
doc = nlp(data[1]['text'])
e = doc.ents[0]
Span(e.text.strip(), e.start_char, e.end_char, e.label_)
```

![image](https://user-images.githubusercontent.com/46970976/107558818-3fdc9d80-6ba9-11eb-8854-1d374937269d.png)

**Reply by svlandeg** [reply]
You can find the `Span` API here: https://spacy.io/api/span

The constructor's first argument should be the `doc`.

**Reply by yw2903** [reply]
Got it. Suppose there is overlap between existing spans and the new spans I set, would there be conflicts?

**Reply by svlandeg** [reply]
If you call `doc.set_ents`, the old set of entities will be entirely replaced by the new set you're giving. So if you process them correctly by keeping the good ones, changing some others but making sure that you don't create overlapping ones, then you should be good.
